projects/npm-tools/packages/npm-scripts/src/utils/tweakWebpackConfig.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * SPDX-FileCopyrightText: © 2019 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const path = require('path');
+const {
+	container: {ModuleFederationPlugin},
+} = require('webpack');
+
+const createTempFile = require('./createTempFile');
+const getMergedConfig = require('./getMergedConfig');
+const parseBnd = require('./parseBnd');
+
+const CORE_REMOTES = ['frontend-js-react-web'];
+const CORE_SHARES = ['react'];
+
+const BABEL_CONFIG = getMergedConfig('babel');
+
+function tweakWebpackConfig(webpackConfig, {federation} = {}) {
+	const ret = [];
+
+	if (webpackConfig) {
+		if (Array.isArray(webpackConfig)) {
+			ret.push(...webpackConfig.map(mergeBabelLoaderOptions));
+		}
+		else {
+			ret.push(mergeBabelLoaderOptions(webpackConfig));
+		}
+	}
+
+	if (federation) {
+		ret.push(createFederationConfig());
+	}
+
+	return ret.length === 1 ? ret[0] : ret;
+}
+
+function createFederationConfig() {
+	// eslint-disable-next-line @liferay/liferay/no-dynamic-require
+	const packageJson = require(path.join(process.cwd(), 'package.json'));
+	const bnd = parseBnd();
+
+	const name = packageJson.name;
+	const main = packageJson.main || 'index.js';
+	const webContextPath = bnd['Web-ContextPath'] || name;
+
+	const {filePath: nullJsFilePath} = createTempFile('null.js', '');
+
+	return {
+		context: process.cwd(),
+		devtool: 'source-map',
+		entry: nullJsFilePath,
+		module: {
+			rules: [
+				{
+					exclude: /node_modules/,
+					test: /\.js$/,
+					use: {
+						loader: 'babel-loader',
+						options: {
+							plugins: [
+								'@babel/plugin-proposal-class-properties',
```

Comment:
Why do we include this specifically?

If we do need to include some extra options here, it would be nice if we could keep this in sync, at least as close as possible, with [our Babel config](https://github.com/liferay/liferay-frontend-projects/blob/de5eb0773d8c19c86e1bcc4ff2a97f0eb024e5f6/projects/npm-tools/packages/npm-scripts/src/config/babel.json#L17-L21) which has a broader list:

```
"@babel/proposal-class-properties",
"@babel/proposal-export-namespace-from",
"@babel/proposal-nullish-coalescing-operator",
"@babel/proposal-object-rest-spread",
"@babel/proposal-optional-chaining",
```

Reply:
True :+1: . I forgot to refactor that.